src/feature/relay/router.h

```diff
@@ -28,7 +28,13 @@ struct ed25519_keypair_t;
 MOCK_DECL(crypto_pk_t *,get_onion_key,(void));
 time_t get_onion_key_set_at(void);
 void set_server_identity_key(crypto_pk_t *k);
+/* Some compilers are clever enough to know that when relay mode is disabled,
+ * this function always asserts. */
+#ifdef HAVE_MODULE_RELAY
 MOCK_DECL(crypto_pk_t *,get_server_identity_key,(void));
+#else
+#define get_server_identity_key() (tor_assert_unreached(),NULL)
```

Comment:
This isn't valid; tor_assert_unreached is a statement, not an expression.

Reply:
Replaced with tor_abort_() in b67c6dad9de3a9b89a367475000c675d7ef7e9b0

Is there a compiler warning that catches issues like this?
clang and gcc both seem very accepting.